contracts/src/rollup/RollupCore.sol

```diff
@@ -400,30 +400,25 @@ abstract contract RollupCore is IRollupCore, PausableUpgradeable {
         {
             uint64 afterInboxPosition = assertion.afterState.globalState.getInboxPosition();
             uint64 prevInboxPosition = assertion.beforeState.globalState.getInboxPosition();
+            // there are 3 kinds of assertions that can be made
+            // 1. ERRORED assertion - in this case not all the messages are processed, and the position in the message can be any
+            // 2. FINISHED assertion that processed all messages - in this case the position in the message is 0, and the after
+            //    state inbox position is the nextInboxPosition of the previous assertion
+            // 3. FINISHED assertion that did not process all messages - since assertions have a fixed size it's possible for an assertion
+            //    to "overflow" when trying to process all the messages in the inbox. In this case the assertionk is in a FINISHED state
+            //    but the number of total messages processed is between 0 and the nextInboxPosition of the previous assertion. The position
+            //    in the message is arbitrary.
+
+            // Since we cannot tell the difference between 2. and 3. without more looking inside the assertion, we can only ensure that
```

Comment:
I think it possible to tell the difference between 2 and 3
(2) would have `prevInboxPosition == configData.nextInboxPosition && prev.posInMsg == 0`
(3) would have `prevInboxPosition < configData.nextInboxPosition || prev.posInMsg > 0`

These can be accessed by `globalState.getInboxPosition()` and `globalState.getInboxPosition()` respectively

Reply:
resolved in slack chat